Add search functionality #18

gemdev111 opened the pull request.

gemdev111 committed:
Refactored ListItemView, added previews with different states
moved ListItemVIew to Lists folder

gemdev111 committed:
moved ListItemProgressView from ListItemView, since it has more then 1 dependecy.  Also used in TransactionsScene separetly

gemdev111 committed:
Project strucuture update
TextStyle moved from Components package to Style package, added necessary Style package imports to Gem
ListItemView minor change

Done #39

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
fixed pbroj, unlinked files after vsc

gemdev111 committed:
Search added for netwroks #18(NetworkSelectorNavigationStack) TODO: - empty state

gemdev111 committed:
added preview for NetworkSelectorNavigationStack

gemdev111 committed:
empty view updated, empty view applied to networkselector empty view

gemdev111 committed:
empty view added as overlay

gemdev111 committed:
updated state view to support exisiting empty view

gemdev111 committed:
removed comments, removed hardcodede localizable

gemdev111 committed:
localisation updates

gemdev111 committed:
added no chains found

gemdev111 committed:
refactor->rename StateEmptyView.message to title

gemdev111 committed:
State empty view updates
- added TextValueView's for easier text handling
- added `footnote`  font if only title injected

gemdev111 committed:
Localisation updates

gemdev111 committed:
ImportWalletType scene updated EmptyState title

gemdev111 committed:
Added search to ChainListSettingsScene

gemdev111 committed:
added constraint if no image and no description change title font to regular

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Merge branch 'main'

Conflicts:
	Packages/Components/Sources/Lists/ListItemView.swift

gemdev111 committed:
removed comments

gemdev111 committed:
Refactored ListItemView & Text Style added defaults value
ListItemView decorated with TextValueView instead of combo text & stle
Added defaults values for TextStyle, Added TextStyle.Preview - to have ability see colors in packages previews

gemdev111 committed:
added note to TextStyle

gemdev111 committed:
Updated Gem dependency with style init

gemdev111 committed:
Project structure update
refactor->rename TextValueView to TextValue
Crated Types folder, moved Value Types(not views) to Types

gemdev111 committed:
Moved System Image to Styles
Added necessary imports Style, removed SystemImage.infoCircle. Since it's repeated image and we don't need it

gemdev111 committed:
Merge branch 'refactor-listItemView'

Conflicts:
	Packages/Components/Sources/SystemImage.swift

gemdev111 committed:
State emptyview updated with TextValueViw

gemdev111 committed:
added necessary import

gemdev111 committed:
added necessary import, updated State empty view with new Style

gemdev111 committed:
Updated init StateEmptyView
#18 DONE

gemdev111 committed:
updated default init of State Empty view with TextStyle static values

gemdev111 committed:
updated initi method with clearer init, added missed values to init

gemdev111 committed:
Merge branch 'refactor-listItemView'

gemcoder21 commented on Gem/Resources/ar.lproj/Localizable.strings:
```diff
@@ -190,4 +190,5 @@
-"stake.minimum_amount" = "الحد الأدنى للمبلغ";
\ No newline at end of file
+"stake.minimum_amount" = "الحد الأدنى للمبلغ";
+"common.no_results_found" = "No Results found";
```
can you rename `No Results found` => `No Results Found` to all cases?

gemdev111 replied:
👌

gemcoder21 commented on Gem/Settings/Navigation/NetworkSelectorNavigationStack.swift:
```diff
@@ -1,35 +1,84 @@
+
     var action: ((String) -> Void)?
 
+    var filteredChains: [Chain] {
```
can you move this into view model?

gemdev111 replied:
Yeah, I would like to) , but there is no way to add vm here without changing a structure of project. If we want to include vm there, we have to rewrite AddTokenNavigationStack, AddTokenScene. That should be a part of separate task

gemcoder21 commented on Gem/Settings/Navigation/NetworkSelectorNavigationStack.swift:
```diff
@@ -1,35 +1,84 @@
+// MARK: - Previews
+
+#Preview {
+    var mockChains = AssetConfiguration.supportedChainsWithTokens
```
maybe just use `Chain.allCases`?

gemdev111 replied:
yeah, make sense

gemcoder21 commented on Gem/Settings/Scenes/ChainListSettings.swift:
```diff
@@ -9,25 +9,42 @@ import Style
                 }
             }
         }
+        .overlay {
```
is there other options besides overlay?

gemdev111 replied:
There are many options to ingrate empty state. The common and modern option to integrate it via overlay, which are best practise for new modern apps iOS16/17+. Like here, https://www.avanderlee.com/swiftui/contentunavailableview-handling-empty-states/ . I defiantly suggest to use it via overlay in that case

gemcoder21 commented on Gem/Settings/ViewModels/ChainListSettingsViewModel.swift:
```diff
@@ -9,8 +9,19 @@ struct ChainListSettingsViewModel {
         return AssetConfiguration.allChains
     }
+
+    func items(for searchText: String) -> [Chain] {
```
this needs to be used with import chains filter, separate search view model should work.

gemcoder21 commented on Packages/Components/Sources/StateView/StateEmptyView.swift:
```diff
@@ -5,17 +5,97 @@ import SwiftUI
+                     .textStyle(title.style)
+                     .multilineTextAlignment(.center)
+
+                 if let description = description {
```
```suggestion
                 if let description {
```

use this style, much shorter

gemcoder21 commented on Packages/Style/Sources/Previews/TextStyle+Preview.swift:
```diff
@@ -0,0 +1,27 @@
+ [Do not use this TextStyle outside previews of Packages, this colors only for shared packages]
+ */
+
+public extension TextStyle {
```
👍

gemcoder21 committed:
Update localize

gemcoder21 commented:
I did update localized strings

gemdev111 committed:
Updated preview mock chain

gemdev111 committed:
if let changed to a new flow

gemdev111 committed:
Merge branch 'main'

Conflicts:
	Gem/Transfer/ViewsModels/ConfirmTransferViewModel.swift

gemdev111 committed:
apr text style updated to correct callout

gemdev111 committed:
Merge remote-tracking branch 'refs/remotes/origin/add-search-functionality'

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Merge branch 'refactor-listItemView'

gemdev111 committed:
Merge branch 'main' into add-search-functionality

gemdev111 committed:
Asset Configuration re-indention

gemdev111 committed:
Added SearchableListView to components
Rename file ChainSettings to ChainSettingsScene
Rename file ChainListSettings to ChainListSettingsScene

gemdev111 committed:
Updates ChainListSettingsScene to support new search
Updates in SearchableListView

gemdev111 committed:
Updates to search flow

SearchableListView update
Added ChainFilterable protocol to omit bipolare code
Updated viewmodels to support new ChainFilterable
Updated NetworkSelectorNavigationStack, added model
AddTokenScene updated ( added address and chain to view model, actions separated in view )

gemdev111 committed:
Merge branch 'main'

Conflicts:
	core

gemcoder21 commented on Gem/Assets/Scenes/AddTokenScene.swift:
```diff
@@ -25,27 +24,29 @@ struct AddTokenScene: View {
                                 .textInputAutocapitalization(.never)
                                 .autocorrectionDisabled()
                                 .submitLabel(.search)
                                 .onSubmit {
```
```suggestion
                               .onSubmit(onSubmitAddress)
```

gemcoder21 commented on Gem/Assets/Scenes/AddTokenScene.swift:
```diff
@@ -88,57 +89,84 @@ struct AddTokenScene: View {
         .navigationTitle(model.title)
         .sheet(isPresented: $isPresentingScanner) {
             ScanQRCodeNavigationStack(isPresenting: $isPresentingScanner) {
```
```suggestion
            ScanQRCodeNavigationStack(isPresenting: $isPresentingScanner, scanResult) {
```

would this work?

gemcoder21 commented on Gem/Assets/Scenes/AddTokenScene.swift:
```diff
@@ -88,57 +89,84 @@ struct AddTokenScene: View {
+    }
 }
 
+// MARK: -
```
can you add marks to all sections, is this an action?

gemdev111 committed:
refactoring ChainFilterable

gemdev111 committed:
Updates AddTokenScene

gemdev111 committed:
ChainListSettingsScene fixed var -> let
AddTokenViewModel added value to MARK
